File: add/codesnippet/VisualBasic/m-system.servicemodel.ch_36_1.vb
		Public Sub ProcessMessage(ByVal input As Message) Implements IUntypedService.ProcessMessage
			Try
				Console.WriteLine("ProcessMessage: Message received: " & input.GetBody(Of String)())
                Dim reqProp = CType(input.Properties(HttpRequestMessageProperty.Name), HttpRequestMessageProperty)
                Dim customString = reqProp.Headers.Get("CustomHeader")
                Dim userAgent = reqProp.Headers(HttpRequestHeader.UserAgent)
				Console.WriteLine()
				Console.WriteLine("ProcessMessage: Got custom header: {0}, User-Agent: {1}", customString, userAgent)
			Catch e As Exception
				Console.WriteLine("ProcessMessage: got exception: " & e.ToString())
			End Try
		End Sub

File: add/codesnippet/CSharp/m-system.servicemodel.ch_36_1.cs
        public void ProcessMessage(Message input)
        {
            try
            {
                Console.WriteLine("ProcessMessage: Message received: " + input.GetBody<string>());
                HttpRequestMessageProperty reqProp = (HttpRequestMessageProperty)input.Properties[HttpRequestMessageProperty.Name];
                string customString = reqProp.Headers.Get("CustomHeader");
                string userAgent = reqProp.Headers[HttpRequestHeader.UserAgent];
                Console.WriteLine();
                Console.WriteLine("ProcessMessage: Got custom header: {0}, User-Agent: {1}", customString, userAgent);
            }
            catch (Exception e)
            {
                Console.WriteLine("ProcessMessage: got exception: " + e.ToString());
            }
        }